### PR TITLE
Add SmoothedUniform distribution

### PR DESCRIPTION
**Pull request by regislebrun**
The random mixture of a Uniform distribution and a Normal distribution is a classical way to model uncertainties resulting from both variability and lack of knowledge. This new distribution is exactly made for that: it is an efficient implementation of such a RandomMixture. As a side effect, it allows for more simplifications in the RandomMixture class.
This PR also provides an improved implementation of RandomMixture::computeScalarQuantile(), with a typical speed-up of 6x~10x when the random mixture is reasonably close to a unimodal distribution. This is the typical situation due to the smoothing effect of the convolution of PDFs.

### CODE REVIEW

File: lib/test/t_SmoothedUniform_std.expout [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+Distribution class=SmoothedUniform name=SmoothedUniform dimension=1 a=-0.5 b=1.5 sigma=0.5
+Distribution SmoothedUniform(a = -0.5, b = 1.5, sigma = 0.5)
+Elliptical = true
+Continuous = true
+oneRealization=class=Point name=Unnamed dimension=1 values=[1.15307]
+oneSample first=class=Point name=Unnamed dimension=1 values=[1.3731] last=class=Point name=Unnamed dimension=1 values=[1.75997]
+mean=class=Point name=Unnamed dimension=1 values=[0.990093]
+covariance=class=CovarianceMatrix dimension=1 implementation=class=MatrixImplementation name=Unnamed rows=1 columns=1 values=[0.581443]
+Kolmogorov test for the generator, sample size=100 is rejected
+Kolmogorov test for the generator, sample size=1000 is rejected
```
Comment (jschueller) [inline lib/test/t_SmoothedUniform_std.expout]: Maybe something wrong with getRealization ?

File: lib/src/Uncertainty/Distribution/SmoothedUniform.cxx [hunk 2]
```diff
@@ -0,0 +1,436 @@
...
+}
+
+/* Parameters value accessor */
+Point SmoothedUniform::getParameter() const
+{
+  Point point(4);
+  point[0] = a_;
+  point[1] = b_;
+  point[2] = sigma_;
+  return point;
```
Comment (jschueller) [inline lib/src/Uncertainty/Distribution/SmoothedUniform.cxx]: Maybe you changed it to 3 parameters ?

File: lib/src/Uncertainty/Distribution/SmoothedUniform.cxx [hunk 3]
```diff
@@ -0,0 +1,436 @@
...
+}
+
+/* Parameters description accessor */
+Description SmoothedUniform::getParameterDescription() const
+{
+  Description description(2);
+  description[0] = "a";
+  description[1] = "b";
+  description[2] = "sigma";
+  return description;
```
Comment (jschueller) [inline lib/src/Uncertainty/Distribution/SmoothedUniform.cxx]: 3 here too

File: lib/test/t_SmoothedUniform_std.expout (hunk 1 above)
Comment (sofianehaddad) [reply lib/test/t_SmoothedUniform_std.expout]: @jschueller agreed. getRealization and getSample differ (missing a_ in getSample)

File: lib/src/Uncertainty/Distribution/SmoothedUniform.cxx [hunk 4]
```diff
@@ -0,0 +1,436 @@
...
+SmoothedUniform * SmoothedUniform::clone() const
+{
+  return new SmoothedUniform(*this);
+}
+
+/* Get one realization of the distribution */
+Point SmoothedUniform::getRealization() const
+{
+  return Point(1, a_ + (b_ - a_) * RandomGenerator::Generate() + sigma_ * DistFunc::rNormal());
+}
```
Comment (sofianehaddad) [inline lib/src/Uncertainty/Distribution/SmoothedUniform.cxx]: Seems that the translation factor a_ should be removed ? compare with getSample/computeMean

File: lib/src/Uncertainty/Distribution/SmoothedUniform.cxx [hunk 5]
```diff
@@ -0,0 +1,436 @@
...
+
+/* Get a sample of the distribution */
+Sample SmoothedUniform::getSample(const UnsignedInteger size) const
+{
+  SampleImplementation result(size, 1);
+  const Point values(RandomGenerator::Generate(size) * (b_ - a_) + DistFunc::rNormal(size) * sigma_);
```
Comment (sofianehaddad) [inline lib/src/Uncertainty/Distribution/SmoothedUniform.cxx]: missing a_

**Comment by regislebrun** [comment]
@sofianehaddad @jschueller Thanks for the review. There are many problems here:
+ a copy/paste from Uniform.cxx for getParameterDescription(), which should be of size 3 and not 2
+ a change of mind concerning SmoothedUniform, which was a 4 parameters distribution at the beginning and ended as a 3 parameters distribution (no need to have a nonzero mean for the Normal component), so the Point of dimension 4 in getParameter() and the missing a_ in getSample (removed when I removed mu_)
I fixed everything on my side but have troubles to interact with github from EDF...

**Comment by sofianehaddad** [comment]
Could you also please add a test to `RandomMixture` that illustrate the changes ? Thanks

**Comment by regislebrun** [comment]
@sofianehaddad The current test already illustrate this change ;-)!

**Comment by jschueller** [comment]
t_Distribution_arithmetic fails on the autobuilder

**Pull request merged**